src/components/CardWrapper.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import { useLocation } from "react-router-dom";
+
+import BingoCard from "./BingoCard.tsx";
+
+import cards from "../assets/cards.json";
+
+function CardWrapper() {
+  const useQuery = () => new URLSearchParams(useLocation().search);
```

Comment:
I quickly tried to move to `useParams` which is more straightforward use, but it wasn't reading the params.